apps/guardian-ui/src/setup/FederationSetup.tsx

```diff
@@ -87,16 +88,38 @@ export const FederationSetup: React.FC = () => {
   let canGoBack = false;
   let canRestart = false;
   let content: React.ReactNode;
+  let roleQueryParam: string | null = null;
 
   switch (progress) {
     case SetupProgress.Start:
       if (needsTosAgreement) {
         title = t('setup.progress.tos.title');
         content = <TermsOfService next={() => setNeedsTosAgreement(false)} />;
       } else {
-        title = t('setup.progress.start.title');
-        subtitle = t('setup.progress.start.subtitle');
-        content = <RoleSelector next={handleNext} />;
+        roleQueryParam = getQueryParam('role');
+        if (roleQueryParam) {
+          roleQueryParam = roleQueryParam.toLowerCase();
+          if (roleQueryParam === 'host') {
+            dispatch({
+              type: SETUP_ACTION_TYPE.SET_ROLE,
+              payload: GuardianRole.Host,
+            });
+          } else if (roleQueryParam === 'solo') {
+            dispatch({
+              type: SETUP_ACTION_TYPE.SET_ROLE,
+              payload: GuardianRole.Solo,
+            });
+          } else if (roleQueryParam === 'follower') {
+            dispatch({
+              type: SETUP_ACTION_TYPE.SET_ROLE,
+              payload: GuardianRole.Follower,
+            });
```

Comment:
Don't call `dispatch` in the render method, render methods should be pure meaning that running them twice should produce the same result, and running a render should not modify application state. These belong in a `useEffect`.